chamge/amqp-arbiter-backend.c

```diff
@@ -32,65 +33,294 @@ G_DEFINE_TYPE (ChamgeAmqpArbiterBackend, chamge_amqp_arbiter_backend, CHAMGE_TYP
 /* *INDENT-ON* */
 
 static ChamgeReturn
-chamge_amqp_arbiter_backend_enroll (ChamgeArbiterBackend * arbiter_backend)
+_amqp_rpc_login (amqp_connection_state_t amqp_conn, amqp_socket_t * amqp_socket,
+    const gchar * url, guint channel)
 {
-  g_autofree gchar *amqp_uri = NULL;
-  g_autofree gchar *amqp_enroll_q_name = NULL;
-  g_autofree gchar *amqp_exchange_name = NULL;
-  gint amqp_channel;
+  amqp_rpc_reply_t amqp_r;
+  struct amqp_connection_info connection_info;
+  g_autofree gchar *str = g_strdup (url);
+
+  amqp_parse_url (str, &connection_info);
+
+  g_return_val_if_fail (connection_info.host != NULL, CHAMGE_RETURN_FAIL);
```

Comment:
Normally, we use assertion for argument to be sure given parameters have right type or value.
The reason to use assertion is to make easy to find wrong type by aborting when G_MESSAGES_DEBUG is enabled.

To let consumer know internal parsing error, we have to consider to change function type to have `GError`.

```

_amqp_rpc_login( ..., GError **err)
{
  ...
  if (connection_info.host == NULL) {
    g_set_error (error, ...);
    return FAILED;
  }
  ...
}